Source/ChatMgr.cpp

```diff
@@ -317,13 +317,12 @@ namespace {
             hour %= 12;
 
 
-        wchar_t* time_buffer = 0;
+        wchar_t time_buffer[29];
         if (localtime.wYear == 0) {
-            time_buffer = Timestamp_seconds ? static_cast<wchar_t[]>(L"[lbracket]--:--:--[rbracket]")
-                                            : static_cast<wchar_t[]>(L"[lbracket]--:--[rbracket]");
+            Timestamp_seconds ? std::memcpy(time_buffer, L"[lbracket]--:--:--[rbracket]", 29)
```

Comment:
`memcpy` takes the size, not the number of characters, so in this case it should be 29*2.